src/common/snippets/include/snippets/lowered/linear_ir.hpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <list>
+
+#include "expression.hpp"
+
+namespace ngraph {
+namespace snippets {
+namespace lowered {
+
+class Config {
+public:
+    // True if the lowered Emitters need to be accessed during runtime. Normally they're destroyed after code emission.
+    bool m_save_lowered_code = false;
+    // True if we should check runtime info for nodes to call specific needed transformations
+    bool m_need_fill_tail_register = false;
+    size_t m_loop_depth = 1;
+};
+
+class LinearIR {
+    class ExpressionFactory;
+public:
+    using container = std::list<ExpressionPtr>;
+    using io_container = std::list<std::shared_ptr<IOExpression>>;
+    using exprIt = container::iterator;
+    using constExprIt = container::const_iterator;
+
+    LinearIR() = default;
+    explicit LinearIR(const std::shared_ptr<ov::Model>& m, Config config = {});
+
+    ExpressionPtr create_expression(const std::shared_ptr<Node>& n, const std::vector<TensorPtr>& inputs);
+
+    static LinearIR::container deep_copy_range(LinearIR::container::const_iterator begin, LinearIR::container::const_iterator end);
+
+    const container& get_ops() const {return m_lowered_ops; }
+    const io_container& get_IO_ops() const {return m_io_lowered_ops; }
```

Comment:
it looks like we are mixing `lowered_ops` and `expressions` naming. Does it make sense to align everywhere?

Reply:
Methods are still named as {smt}_ops. :) 
Ok to postpone fix to seaprate PR.

Reply:
Oh, yeah, I see. My eyes are used to this naming and I didn't notice that. Thank you. I will fix it in separate PR